src/groups/mqb/mqbs/mqbs_filebackedstorage.t.cpp

```diff
@@ -0,0 +1,2010 @@
+           bslma::Allocator*        allocator,
+           bool                     toConfigure = false)
+    : BaseTester(allocator)
+    , d_dataStore(allocator, partitionId)
+    {
+        mqbconfm::Domain domainCfg;
+        domainCfg.deduplicationTimeMs() = 0;  // No history
+        domainCfg.messageTtl()          = ttlSeconds;
+
+        d_replicatedStorage_mp.load(
+            new (*d_allocator_p)
+                mqbs::FileBackedStorage(&d_dataStore,
+                                        bmqt::Uri(uri, s_allocator_p),
+                                        queueKey,
+                                        domainCfg,
+                                        d_mockDomain.capacityMeter(),
+                                        d_allocator_p),
+            d_allocator_p);
+
+        setupReplicatedStorage(ttlSeconds, toConfigure);
+    }
+
+    const MockDataStore& dataStore() { return d_dataStore; }
+};
+
+// ================
+// struct BasicTest
+// ================
+
+/// Fixture instantiating a tester of `mqbs::FileBackedStorage` having not yet
+/// configured the storage with a storage configuration.
+struct BasicTest : mwctst::Test {
+    // PUBLIC DATA
+    Tester d_tester;
+
+    // CREATORS
+    BasicTest();
+    ~BasicTest() BSLS_KEYWORD_OVERRIDE;
+};
+
+// ----------------
+// struct BasicTest
+// ----------------
+
+// CREATORS
+BasicTest::BasicTest()
+: d_tester(k_URI_STR,
+           k_QUEUE_KEY,
+           k_PROXY_PARTITION_ID,
+           k_INT64_MAX,  // ttlSeconds
+           s_allocator_p,
+           false)  // toConfigure
+{
+    // NOTHING
+}
+
+BasicTest::~BasicTest()
+{
+    // NOTHING
+}
+
+// ===========
+// struct Test
+// ===========
+
+/// Fixture instantiating a tester of `mqbs::FileBackedStorage` having already
+/// configured the storage with an FileBackedStorage configuration.
+struct Test : mwctst::Test {
+    // PUBLIC DATA
+    Tester d_tester;
+
+    // CREATORS
+    Test();
+    ~Test() BSLS_KEYWORD_OVERRIDE;
+};
+
+// -----------
+// struct Test
+// -----------
+// CREATORS
+Test::Test()
+: d_tester(k_URI_STR,
+           k_QUEUE_KEY,
+           k_PARTITION_ID,
+           k_INT64_MAX,  // ttlSeconds
+           s_allocator_p,
+           true)  // toConfigure
+{
+    // NOTHING
+}
+
+Test::~Test()
+{
+    // NOTHING
+}
+
+// =============
+// struct GCTest
+// =============
+
+/// Fixture instantiating a tester of `mqbs::FileBackedStorage` instantiated
+/// with parameterized `TTL` and having already configured the storage with
+/// an FileBackedStorage configuration.
+struct GCTest : mwctst::Test {
+    // PUBLIC DATA
+    bsls::ObjectBuffer<Tester> d_testerBuffer;
+
+    // CREATORS
+    GCTest();
+    ~GCTest() BSLS_KEYWORD_OVERRIDE;
+
+    // MANIPULATORS
+    void configure(bsls::Types::Int64 ttlSeconds);
+
+    Tester& tester();
+};
+
+// -------------
+// struct GCTest
+// -------------
+// CREATORS
+GCTest::GCTest()
+: d_testerBuffer()
+{
+    // NOTHING
+}
+
+GCTest::~GCTest()
```

Comment:
Actually, I am thinking if we have too much test related classes here. I don't see the point of creating so many of them.
What we have:
`mwctst::Test`, `GCTest`, `Test`, `BasicTest`
Can we only have 2 of them? Just `mwctst::Test` and `Test`?

The same thing with testers classes.
We have: `Tester`, `BaseTester`.
Can we have only one `Tester`?

I have a feeling that we are overcomplicating things by this bloated class structure

Reply:
Done. Although maybe there is now more test code duplication, but the code of the unit tests became more straightforward. 

Please take a look. And if everything is ok, I'll transfer the changes to InMemoryStorage unit tests too

Reply:
Migrated the refactoring to the InMemoryStorage unit tests